joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -91,6 +91,7 @@ trait TypedScope[M <: MethodLike, F <: FieldLike, T <: TypeLike[M, F]](summary:
   def tryResolveTypeReference(typeName: String): Option[T] = {
     typesInScope
       .collectFirst {
+        case typ if typ.name.split("[.]").lastOption.getOrElse("").equals(typeName)       => typ
         case typ if typ.name.endsWith(typeName)                                           => typ
```

Comment:
This check was still needed incase we encounter `fullyQualifiedName`.

Reply:
Ah wait, if you always assume `typeName` is not qualified then yes, but if you make the first check:
```scala
typ.name.split("[.]").lastOption == typeName.split("[.]").lastOption
```
then it should work, but we would want to also eventually want some way to match something like
`Foo.Bar.Baz` with `Bar.Baz` (partially qualified if `using Bar;` was qualified at the top. So for that we would want to look for some intersection of the suffixing parts, but let's leave a TODO for now